parser/src/lexer/tests.rs

```diff
@@ -45,11 +45,16 @@ fn keywords_with_identifiers() {
 
 #[test]
 fn multi_arithmetic_ops() {
-    let source = "enf clk' - clk - 1 = 0";
+    let source = "enf clk' * clk - clk' - clk - 1 = 0";
```

Comment:
I've renamed these to `chained_<op>_name` but I'm not sure if there's much utility in doing extensive testing for lexer for chained operations since these should only cause error at the parser level. Maybe all of lexer tests should be structured differently than I've done in the updated PR. Please let me know your thoughts on this.

Reply:
I think you're right about most of the complex tests being at the parser level. We can audit the tests later to assess which edge cases we're missing and which tests are redundant. For now, I think what you have is good, and I like the restructure.